corehq/ex-submodules/pillowtop/checkpoints/manager.py

```diff
@@ -14,82 +14,81 @@
 DocGetOrCreateResult = namedtuple('DocGetOrCreateResult', ['document', 'created'])
 
 
-class PillowCheckpointManager(object):
-
-    def __init__(self, dao):
-        self._dao = dao
-
-    def get_or_create_checkpoint(self, checkpoint_id):
-        created = False
-        try:
-            checkpoint_doc = self._dao.get_document(checkpoint_id)
-        except DocumentNotFoundError:
-            checkpoint_doc = {'seq': '0', 'timestamp': get_formatted_current_timestamp()}
-            self._dao.save_document(checkpoint_id, checkpoint_doc)
-            created = True
-        return DocGetOrCreateResult(checkpoint_doc, created)
+def get_or_create_checkpoint(checkpoint_id):
+    created = False
+    try:
+        checkpoint = DjangoPillowCheckpoint.objects.get(checkpoint_id=checkpoint_id)
+    except DjangoPillowCheckpoint.DoesNotExist:
+        checkpoint = DjangoPillowCheckpoint.objects.create(
+            checkpoint_id=checkpoint_id,
+            sequence='0',
+            timestamp=datetime.utcnow(),
+        )
+        created = True
+    return DocGetOrCreateResult(checkpoint, created)
 
-    def reset_checkpoint(self, checkpoint_id):
-        checkpoint_doc = self.get_or_create_checkpoint(checkpoint_id).document
-        checkpoint_doc['old_seq'] = checkpoint_doc['seq']
-        checkpoint_doc['seq'] = '0'
-        checkpoint_doc['timestamp'] = get_formatted_current_timestamp()
-        self._dao.save_document(checkpoint_id, checkpoint_doc)
 
-    def update_checkpoint(self, checkpoint_id, checkpoint_doc):
-        self._dao.save_document(checkpoint_id, checkpoint_doc)
+def reset_checkpoint(checkpoint_id):
+    checkpoint = get_or_create_checkpoint(checkpoint_id).document
+    checkpoint.old_sequence = checkpoint.sequence
+    checkpoint.sequence = '0'
+    checkpoint.timestamp = datetime.utcnow()
+    checkpoint.save()
 
 
 class PillowCheckpoint(object):
 
-    def __init__(self, dao, checkpoint_id):
-        self._manager = PillowCheckpointManager(dao=dao)
+    def __init__(self, checkpoint_id):
         self.checkpoint_id = checkpoint_id
         self._last_checkpoint = None
 
     def get_or_create(self, verify_unchanged=False):
-        result = self._manager.get_or_create_checkpoint(self.checkpoint_id)
+        result = self._get_or_create(verify_unchanged)
+        return DocGetOrCreateResult(DjangoPillowCheckpoint.to_dict(result.document), result.created)
```

Comment:
no - was just easier to keep backwards compatible. i got rid of the `dict` version in https://github.com/dimagi/commcare-hq/pull/10408